File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/routines/InoutProcedure.scala
package dev.typr.foundations.docs.routines
import dev.typr.foundationssc.*

import java.sql.SQLException

@SuppressWarnings(Array("unused"))
object InoutProcedure:
  private val tx: Transactor = null // placeholder

  //start
  // INOUT — the value goes in and comes back modified
  val applyDiscount =
    DbProcedure.define("apply_discount")
      .input(PgTypes.text)       // discount_code IN
      .inout(PgTypes.numeric)   // price INOUT
      .build()

  @throws[SQLException]
  def applyDiscount(code: String, price: BigDecimal): BigDecimal =
    applyDiscount.call(code, price).transact(tx)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/routines/InoutProcedure.java
package dev.typr.foundations.docs.routines;

import dev.typr.foundations.*;
import java.math.BigDecimal;

@SuppressWarnings("unused")
public class InoutProcedure {
  Transactor tx = null; // placeholder

  // start
  // INOUT — the value goes in and comes back modified
  static final DbProcedure.Def2_1<String, BigDecimal, BigDecimal> applyDiscount =
      DbProcedure.define("apply_discount").input(PgTypes.text).inout(PgTypes.numeric).build();

  BigDecimal applyDiscount(String code, BigDecimal price) {
    return applyDiscount.call(code, price).transact(tx);
  }
  // stop
}
